concepts/array/about.md

```diff
@@ -0,0 +1,186 @@
+# Array
+
+[Array][array] is a mutable data structure that stores a collection of elements of a specific type.
+An array is an [indexable][indexable] data structure.
+Arrays being mutable means that if a method is called on an array that modifies the array, the array will be modified.
+Meaning it doesn't have to be reassigned to the variable.
+
+To create an array, use the array literal denotation syntax (`[]`) and within it, specify the elements of the array separated by a comma.
+
+```crystal
+[1, 2, 3] # => [1, 2, 3]
+```
+
+Crystal will infer the type of the array from the elements.
+
+```crystal
+[1, 2, 3].class # => Array(Int32)
+```
+
+## Multi type Arrays
+
+Even if mentioned earlier about arrays being a collection of elements of a specific type, you can create an array with multiple types through the use of [union types][union-types].
+This makes so that the array can contain elements of any of the types specified in the union type.
+Crystal will infer the type of the array from the elements.
+
+```crystal
+[1, "2", 3.0] # => [1, "2", 3.0]
+
+[1, "2", 3.0].class # => Array(Int32 | Float64 | String)
+```
+
+## Add an element to an array
+
+To add an element to an array, use the [`<<` (append) operator][append].
+
+```crystal
+[1, 2, 3] << 4 # => [1, 2, 3, 4]
+```
+
+It is important that the type of the element you want to add is the same as the type of the array, if it is not an error will be raised.
+
+```crystal
+numbers : Array(Int32 | Float64) = [1, 2, 3]
+numbers << 4.0
+numbers # => [1, 2, 3, 4.0]
+numbers << "5" # => Error: no overload matches 'Array(Int32 | Float64)#<<' with type String
+```
+
+## Size of an Array
+
+As with `String`, can you get the size of an array by using the [`size`][size] method.
+
+```crystal
+[1, 2, 3].size # => 3
+```
+
+## Empty Arrays
+
+When creating an empty array, can the compiler not infer which type the array is built of.
+Therefore, you need to specify the type of the array.
+This can be done by either specifying the type of the array or by using the `of` keyword.
+Or by using the array initializer syntax, which is: `Array(T).new`.
```

Comment:
```suggestion
This can be done by specifying the type of the array, by using the `of` keyword, or by using the array initializer syntax, which is: `Array(T).new`.
```